distributed/shuffle/_worker_plugin.py

```diff
@@ -951,7 +951,7 @@ def split_by_worker(
     # bytestream such that it cannot be deserialized anymore
     t = pa.Table.from_pandas(df, preserve_index=True)
     t = t.sort_by("_worker")
-    codes = np.asarray(t.select(["_worker"]))[0]
+    codes = np.asarray(t["_worker"])
```

Comment:
So before you were converting a single-column Table object to a 2D ndarray of shape `(1, n)` and then getting the 1D array of shape `(n, )` with that indexing operation. 
With latest pyarrow, that gives shape `(n, 1)`, and then `[0]` no longer does the correct thing. 

But we can also directly convert the column to a numpy array, and then we directly get a 1D array.